### ISSUE TITLE
Feature: add verification of mining pool input address

### ISSUE THREAD

**Issue by lwisne** [iron-fish/ironfish#2039]
### Description

We have seen a handful of this error in the pool logs:

```
Error while sending payout transaction: Request failed (400) error: InvalidPaymentAddress
Error: InvalidPaymentAddress
    at CreateTransactionTask.execute (/usr/src/app/node_modules/@ironfish/sdk/build/src/workerPool/tasks/createTransaction.js:157:26)
    at Object.handleRequest (/usr/src/app/node_modules/@ironfish/sdk/build/src/workerPool/tasks/handlers.js:30:20)
    at Job.execute (/usr/src/app/node_modules/@ironfish/sdk/build/src/workerPool/job.js:58:25)
    at MessagePort.Worker.onMessageFromParent (/usr/src/app/node_modules/@ironfish/sdk/build/src/workerPool/worker.js:63:18)
    at MessagePort.[nodejs.internal.kHybridDispatch] (node:internal/event_target:643:20)
    at MessagePort.exports.emitMessage (node:internal/per_context/messageport:23:28)
```

It seems likely that this happened because the command `ironfish miners:start --pool pool.ironfish.network --address <PUBLIC KEY>` was given an invalid public key.

If this happens we should catch the error and bail from the miner command with an appropriate error message. Slightly better, but a bit more complicated, would be to check right off the bat when the command is started.

**Comment by lwisne** [comment]
We actually check the length and format of the public address early on, so it's OK to just catch this error in the much rarer case that someone uses a correctly formatted address that isn't the public address for an account

**Comment by hairtail** [comment]
> We actually check the length and format of the public address early on, so it's OK to just catch this error in the much rarer case that someone uses a correctly formatted address that isn't the public address for an account

It's hard to check whether a public address is from an account. In most cases, we just check the length and format and maybe checksum of the byte array.

**Comment by hughy** [comment]
Resolved by #2040
